src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/SignatureHelper.cs

```diff
@@ -223,12 +223,76 @@ private static void WriteSignatureForType(SignatureTypeEncoder signature, Type t
             {
                 signature.GenericTypeParameter(type.GenericParameterPosition);
             }
+            else if (type.IsFunctionPointer)
+            {
+                WriteSignatureForFunctionPointerType(signature, type, module);
+            }
             else
             {
                 WriteSimpleSignature(signature, type, module);
             }
         }
 
+        private static void WriteSignatureForFunctionPointerType(SignatureTypeEncoder signature, Type type, ModuleBuilderImpl module)
+        {
+            SignatureCallingConvention callConv = SignatureCallingConvention.Default;
+            FunctionPointerAttributes attribs = FunctionPointerAttributes.None;
+
+            Type returnType = type.GetFunctionPointerReturnType();
+            Type[] paramTypes = type.GetFunctionPointerParameterTypes();
+
+            if (type.IsUnmanagedFunctionPointer)
+            {
+                callConv = SignatureCallingConvention.Unmanaged;
+
+                if (type.GetFunctionPointerCallingConventions() is Type[] conventions && conventions.Length == 1)
+                {
+                    switch (conventions[0].FullName)
+                    {
+                        case "System.Runtime.CompilerServices.CallConvCdecl":
```

Comment:
When we replace the modopt with callConv, the modopt needs to get deleted from modotps that get emitted (ie from the list returned by GetOptionalCustomModifiers and passed into WriteCustomModifiers).

Reply:
The test that I have shared at https://github.com/dotnet/runtime/pull/119935#discussion_r2446337717 shows that `CallConvCdecl` shows up in `GetOptionalCustomModifiers` for `delegate* unmanaged<CDecl>`. Did I make a mistake somewhere?

Reply:
Sorry, my mistake. Thank you for pointing it out